### PR TITLE
feat(unlock-app): Add redesigned keychain and subscription info

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

CleanShot 2022-12-01 at 23 54 46@2x
CleanShot 2022-12-02 at 00 05 32@2x

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
I like the general idea a lot! I think the "valid" label could be displayed in a different location maybe?

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 1]
```diff
@@ -201,49 +243,64 @@ function Key({ ownedKey, account, network }: Props) {
...
     const url = opensea?.tokenUrl(lock.address, tokenId) ?? null
-
-    if (url) {
-      window.open(url, '_blank')
-    } else {
-      ToastHelper.error(`OpenSea URL not unavailable yet for ${name}`)
+    if (!url) {
+      return
     }
+    window.open(url)
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:268]: Let's open external links in new tabs please!

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 2]
```diff
@@ -201,49 +243,64 @@ function Key({ ownedKey, account, network }: Props) {
...
   }
 
-  async function exploreLock() {
-    window.open(networks[network].explorer?.urls.address(lock.address))
+  const onExploreLock = () => {
+    const url = networks[network].explorer?.urls.address(lock.address)
+    if (!url) {
+      return
+    }
+    window.open(url)
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:259]: Let's open external links in new tabs please!

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 3]
```diff
@@ -302,76 +351,122 @@ function Key({ ownedKey, account, network }: Props) {
...
+              <div className="p-1">
+                <Menu.Item>
+                  {({ disabled, active }) => (
+                    <MenuButton
+                      disabled={disabled}
+                      active={active}
+                      onClick={onExploreLock}
+                    >
+                      <ExploreIcon size={16} />
+                      Explore lock
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:388]:
```suggestion
                      Block explorer
```

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 4]
```diff
@@ -113,17 +133,39 @@ const KeyBox = ({
...
-            </p>
-          </>
+          <p className="flex items-center justify-between gap-2 text-sm">
+            <span className="text-gray-500">Valid</span>
+            <span className="font-medium">{expirationStatus}</span>
+          </p>
+        )}
+        {nextRenewal && (
+          <p className="flex items-center justify-between gap-2 text-sm">
+            <span className="text-gray-500">Renew on</span>
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:148]:
```suggestion
            <span className="text-gray-500">Renews on</span>
```

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 5]
```diff
@@ -113,17 +133,39 @@ const KeyBox = ({
...
+            <span className="font-medium">{nextRenewal}</span>
+          </p>
+        )}
+        {approvedTime && (
+          <p className="flex items-center justify-between gap-2 text-sm">
+            <span className="text-gray-500">Renew for</span>
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:154]:
```suggestion
            <span className="text-gray-500">Renews for</span>
```

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 6]
```diff
@@ -54,52 +69,57 @@ const KeyBox = ({
   const [isCopied, setCopied] = useClipboard(lock.address, {
     successDuration: 2000,
   })
+
+  // Replace with actual call to backend
+  const nextRenewal = '9 PM : 12 March 2022'
+  const userBalance = '12 LINK'
+  const approvedTime = '12 days'
+  const type = 'crypto'
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:77]: I am not sure we have an easy way to distinguish things between crypto/card/grants... etc so Probably better to skip!

**Comment by ChialiTsai** [comment]
image

I'd like to propose some adjustment, see detail here: https://www.figma.com/file/pNN7ER1F4mdToFfk1bB0gu?node-id=2904:79214#318932430

File: unlock-app/src/components/interface/keychain/Key.tsx (hunk 6 above)
Comment (searchableguy) [reply unlock-app/src/components/interface/keychain/Key.tsx]: The subscription endpoint tries to guess based on what information we have.

**Comment by julien51** [comment]
I am not a huge fan of all the lines... but I guess that's a question of taste :D Let me look at the code!

File: unlock-app/src/components/interface/keychain/Key.tsx [hunk 7]
```diff
@@ -201,49 +259,66 @@ function Key({ ownedKey, account, network }: Props) {
...
+  const isExtendable = lock?.version >= 11
 
-  const viewOnOpenSea = async () => {
-    const { opensea, name } = networks[network]
+  const onExploreLock = () => {
+    const url = networks[network].explorer?.urls.address(lock.address)
+    if (!url) {
+      return
+    }
+    window.open(url, '_')
```
Comment (julien51) [inline unlock-app/src/components/interface/keychain/Key.tsx:277]: thx!

**Review by julien51 — APPROVED** [review]
LGTM

**Pull request merged**